### DISCUSSION TITLE
mmcls 支持 base64 数据输入嘛🐎？

### DISCUSSION THREAD

**Discussion by TPL-egg-HIT** [open-mmlab/mmclassification#1574] [Q&A]
mmcls支持base64数据作为输入嘛，我看了官方文档，也找到了如下的代码
mmcv/fileio/file_client.py
我的结论是目前不支持base64作为输入，那我需要怎么修改代码呢，谢谢～

**Comment by TPL-egg-HIT** [comment]
请救救孩子😭

**Comment by Ezra-Yu** [comment]
You can modify the `LoadImageFromFile` transformer. use base64 to decode the stream. notice this [line](https://github.com/open-mmlab/mmcv/blob/0c23eb0215f8da679513a8ec8fcbfc6de9cda3bf/mmcv/transforms/loading.py#L101). 
```
import base64

img = base64.b64decode((img_bytes))
```

fro the detail, please refer to https://www.codespeedy.com/convert-base64-string-to-image-in-python/

**Reply by TPL-egg-HIT** [reply]
THANKS~
It seems that, I need to define a new func to support base64,
and change 
        try:
            if self.file_client_args is not None:
                file_client = fileio.FileClient.infer_client(
                    self.file_client_args, filename)
                img_bytes = file_client.get(filename)
            else:
                img_bytes = fileio.get(
                    filename, backend_args=self.backend_args)
            img = mmcv.imfrombytes(
                img_bytes, flag=self.color_type, backend=self.imdecode_backend)
        except Exception as e:
to : 
        try:
            img = mmcv.imfrombytes(
                img_bytes, flag=self.color_type, backend=self.imdecode_backend) # imdecode_backend - I need to define a new name.
        except Exception as e:

**Reply by Ezra-Yu** [reply]
just replace this line :
```

img = mmcv.imfrombytes(img_bytes, flag=self.color_type, backend=self.imdecode_backend)
```
